pahfit/helpers.py

```diff
@@ -0,0 +1,138 @@
+import os
+import pkg_resources
+
+import astropy.units as u
+from astropy.table import Table
+
+from astropy.modeling.fitting import LevMarLSQFitter
+
+from pahfit.base import PAHFITBase
+
+__all__ = ["read_spectrum", "initialize_model", "fit_spectrum"]
+
+
+def read_spectrum(specfile):
+    """
+    Read in a spectrum and convert intput units to the expected internal PAHFIT units.
+
+    Parameters
+    ----------
+    specfile : string
+        file with the spectrum to be fit
+
+    Returns
+    -------
+    obsdata : dict
+        x is wavelength in microns and y/unc are the spectrum/unc in units
+        of Jy
+    """
+
+    # read in the observed spectrum
+    # assumed to be astropy table compatibile and include units
+    specfile = specfile
+    if not os.path.isfile(specfile):
+        pack_path = pkg_resources.resource_filename("pahfit", "data/")
+        test_specfile = "{}/{}".format(pack_path, specfile)
+        if os.path.isfile(test_specfile):
+            specfile = test_specfile
+        else:
+            raise ValueError("Input spectrumfile {} not found".format(specfile))
+
+    # get the table format (from extension of filename)
+    tformat = specfile.split(".")[-1]
+    if tformat == "ecsv":
+        tformat = "ascii.ecsv"
+    obs_spectrum = Table.read(specfile, format=tformat)
+    obsdata = {}
+    obsdata["x"] = obs_spectrum["wavelength"].to(u.micron, equivalencies=u.spectral())
+    obsdata["y"] = obs_spectrum["flux"].to(
+        u.Jy, equivalencies=u.spectral_density(obsdata["x"])
+    )
+    obsdata["unc"] = obs_spectrum["sigma"].to(
```

Comment:
By using "wavelength", "flux" and "sigma" here we imply that the input spectrum's must have the exact column naming in the header, otherwise it will raise an error. Perhaps we can allow more freedom by retrieving the data column names as provided by the user, using colnames (colnames[0], colnames[1], and colnames[2)]. This way we don't force a naming convention, instead we only request to have the wavelength, flux, and uncertainty columns in that order, and we can specify it in the docs.

Reply:
Good idea.  Will implement as a keyword to this function with the current as the default.